sklearn/cross_decomposition/_pls.py

```diff
@@ -295,6 +295,18 @@ def fit(self, X, Y):
 
         self.coef_ = np.dot(self.x_rotations_, self.y_loadings_.T)
         self.coef_ = self.coef_ * self.y_std_
+
+        if self.deflation_mode == "regression":
+            # Calculate VIP for PLSRegression
+            T = self.x_scores_
+            W = self.x_weights_
+            Q = self.y_loadings_
+            w0, w1 = W.shape
+            s = np.sum(T ** 2, axis=0) * np.sum(Q ** 2, axis=0)
+            s_sum = np.sum(s, axis=0)
+            w_norm = np.array([(W[:, i] / np.linalg.norm(W[:, i]))
+                               for i in range(w1)])
+            self.vip_ = np.sqrt(w0 * np.sum(s * w_norm.T ** 2, axis=1) / s_sum)
```

Comment:
I think this could be written more compact to get rid of the for loop as below.
`self.vip_ = np.sqrt(w0 * np.sum(s * (W / np.linalg.norm(W, axis=0)) ** 2, axis=1) / s_sum)`